src/Realm.C

```diff
@@ -2954,7 +2955,29 @@ Realm::overset_orphan_node_field_update(
   const unsigned sizeRow,
   const unsigned sizeCol)
 {
+#ifdef KOKKOS_ENABLE_CUDA
+  throw std::runtime_error("Non-NGP version of overset algorithm called in NGP build");
+#else
   oversetManager_->overset_update_field(theField, sizeRow, sizeCol);
+#endif
+}
+
+void
+Realm::overset_field_update(
+  stk::mesh::FieldBase* field,
+  const unsigned nRows,
+  const unsigned nCols,
+  const bool doFinalSyncToDevice)
+{
+  const auto& fieldMgr = ngp_field_manager();
+  auto& ngpField =
+    fieldMgr.get_field<double>(field->mesh_meta_data_ordinal());
+  ngpField.sync_to_host();
+  oversetManager_->overset_update_field(field, nRows, nCols);
+  ngpField.modify_on_host();
+
+  if (doFinalSyncToDevice)
+    ngpField.sync_to_device();
```

Comment:
I think a guiding convention i would like to advocate is keeping the modify_on_host/modify_on_device calls as close as possible to the code that is doing the modifying, and keeping the sync_to_host/sync_to_device calls as close as possible to the code that needs the sync'd data.

Reply:
See https://github.com/Exawind/nalu-wind/pull/591#discussion_r426326776